x-pack/plugins/cases/public/components/case_action_bar/index.tsx

```diff
@@ -68,19 +66,7 @@ const CaseActionBarComponent: React.FC<CaseActionBarProps> = ({
     [onUpdateField]
   );
 
-  const { data: userActionsData, isLoading: isLoadingUserActions } = useGetCaseUserActions(
-    caseData.id,
-    caseData.connector.id
-  );
-
-  const currentExternalIncident = useMemo(
-    () =>
-      userActionsData?.caseServices != null &&
-      userActionsData.caseServices[caseData.connector.id] != null
-        ? userActionsData.caseServices[caseData.connector.id]
-        : null,
-    [userActionsData?.caseServices, caseData.connector]
-  );
+  const currentExternalIncident = caseData.externalService ?? null;
```

Comment:
There is no need to get the external service information from the user actions. We can get it from the case data. The `currentExternalIncident` is used to show the "View <external_incident_name>" action which is always the latest push.

<img width="229" alt="Screenshot 2023-01-23 at 2 25 30 PM" src="https://user-images.githubusercontent.com/7871006/214039111-0d4692c3-e7c3-4b7b-825b-bd86a0f03a96.png">

Reply:
This is a bug. I will fix it when @jonathan-buttner PR is merged. We should extend the API to return the `externalService` (payload) of the latest push.

Reply:
For reference the PR is here: https://github.com/elastic/kibana/pull/149535

Reply:
Fixed in https://github.com/elastic/kibana/pull/149276/commits/54963b3c8579c0e4267d315b3bb6c82fd90cd1b4